### DISCUSSION TITLE
Help: Interrupt on Preamble Detected?

### DISCUSSION THREAD

**Discussion by jomar0** [jgromes/RadioLib#469] [Q&A]
Hey guys,
I was hoping someone could help me with interrupts using the RFM95. I am building a low power LoRa jammer for my University Project.
My ESP32 based device needs to wake from deep-sleep when a preamble is detected, but I'm not sure how to go about this.

If anyone can help, or point me in the direction of someone who can, I would greatly appreciate it.

Thanks,
Jomar

**Comment by jgromes** [comment]
As far as I can tell, SX127x/RFM95 will not be able to reliably generate an interrupt based on presence of LoRa preamble. It can generate an interrupt during channel activity detection (CAD), but IIRC, just preamble is not enough. You also can't use simple measures like the RSSI, since LoRa signals can be below the noise floor.

Maybe SX126x would be a better choice, as it can detect just the preamble in CAD mode.

**Reply by jomar0** [reply]
At the moment, I have used the scanChannel() function to check if the channel is active. I can probably use scanChannel after waking up to make sure the channel is active before broadcasting the jamming message. I will likely test reliability both with and without.

How would I go about DIO mapping and editing the memory-mapped registers in general? There is a section on page 109 of the RFM95 TRM that describes how to wake on preamble which I'll base my implementation on. I've only ever changed the registers on the ESP32 device.

Thanks so much for your help!

**Reply by jgromes** [reply]
I believe that section only applies to FSK mode, not LoRa.

You can use the interrupt-based version of CAD, then use that interrupt to wake up your ESP32.

I'm a bit sceptical about the project in general to be honest, LoRa is notoriously resilient to in-band interference. You would also have to know in advance which spreading factor are you trying to jam, since you can transmit in the same band with different SFs overlapping, and the receiver should have no trouble picking them apart.

**Reply by jomar0** [reply]
I know about the different SF thing, but I have had success with the device already, on the same SF, obviously. But there aren't many available SFs anyway.
I'll give the CAD interrupt a go.
Thanks
